src/tfr_control/src/arm_action_server.cpp

```diff
@@ -0,0 +1,103 @@
+/****************************************************************************************
+ * File:    arm_action_server.cpp
+ * Node:    arm_action_server
+ * 
+ * Purpose: This ActionServer is what handles the movement of the arm to given points.
+ *          It's tasked with interfacing with the move_group node to execute these
+ *          actions.
+ * 
+ *          This file includes <tfr_msgs/Arm.h>, which is one of seven headers
+ *          built by catkin from `tfr_msgs/action/Example.action`:
+ *              devel/include/tfr_msgs/ExampleAction.h
+ *              devel/include/tfr_msgs/ExampleActionFeedback.h
+ *              devel/include/tfr_msgs/ExampleActionGoal.h
+ *              devel/include/tfr_msgs/ExampleActionResult.h
+ *              devel/include/tfr_msgs/ExampleFeedback.h
+ *              devel/include/tfr_msgs/ExampleGoal.h
+ *              devel/include/tfr_msgs/ExampleResult.h
+ * 
+ ***************************************************************************************/
+
+#include <actionlib/server/simple_action_server.h>
+#include <tfr_msgs/ArmMoveAction.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+
+typedef actionlib::SimpleActionServer<tfr_msgs::ArmMoveAction> Server;
+
+moveit::planning_interface::MoveGroupInterface *move_group;
+const robot_state::JointModelGroup *joint_model_group;
+
+// This is the method that will be called when a client makes use
+// of this server. The provided goal is the "input"
+void execute(const tfr_msgs::ArmMoveGoalConstPtr& goal, Server* server)
+{
+    // Set up the joint space goal vector to travel to based on the input goal
+    // from the action server
+    std::vector<double> joint_group_positions(4);
+    joint_group_positions[0] = goal->pose[0];
+    joint_group_positions[1] = goal->pose[1];
+    joint_group_positions[2] = goal->pose[2];
+    joint_group_positions[3] = goal->pose[3];
+
+    // Set the current target
+    move_group->setJointValueTarget(joint_group_positions);
+    moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+
+    // Try to plan to the given target
+    bool success = (move_group->plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+
+    if (success) {
+        // Planning was successful, actually execute the movement
+        ROS_DEBUG("Executing movement");
+        move_group->move();
+        // TODO: How are errors handled?
+        //  - Further note: in the case that, say, the arm gets stuck and can't
+        //    move, how are these errors returned? Exceptions, return code, or
+        //    something else?
```

Comment:
You have a comment that asks how errors are handled.  Are you asking how the `move` function handles errors, or how we want to handle them?

Reply:
I'm asking how the move function handles errors: I know how I want to handle them I think, it'll just be returning a failure and maybe stopping execution of the arm command.

Reply:
... took me literally 30 seconds to find the answer to this, so I'm good now. Added a quick super simple implementation of it, incoming in like 5 minutes.